osprey_flask_app/utils.py

```diff
@@ -59,3 +63,48 @@ def create_full_arg_dict(args):
 
             arg_dict[arg] = args.get(arg, default=default)
     return arg_dict
+
+
+def validate_inputs(arg_dict):
```

Comment:
While it's nice to delegate the validity checking to a separate function, the calling relationship is a little funky. Like, why do you return an HTTP response, if you're only going to actually use the HTTP response in the exceptional case (if the inputs are not valid).

I see that you want to return information about *why* the input is not valid, so that you can return it to the user. So that's good. Maybe we could rename the function to: `inputs_are_valid` and it either returns `True` or raises an exception with the message.

Then the calling function could read:

```python3
try:
    # The if clause is superfluous, but it aids readability to some extent
    if inputs_are_valid(...):
        # return 200
except Exception as e:
     return Response(e.message, status=400)
```

Maybe? @nikola-rados thoughts?